### ISSUE TITLE
503 or Requested Range Not Satisfiable with 0Kb file size

### ISSUE THREAD

**Issue by acsfer** [nextcloud/server#22029]
- NC 19.0.1
- Object Storage as primary

## Issue
- Upload this file to NC with the Browser to the root: https://hiberfile.com/d/Yza9WDIA
- Try to sync with Nextcloud Desktop client (Win) or download (by clicking on it)move/copy it into another folder using the WebUI.

It will produce either:

`fopen(urn%3Aoid%3A18377): failed to open stream: HTTP request failed! HTTP/1.1 416 Requested Range Not Satisfiable`

on `nextcloud.log` or

`503 Service Temporarily Unavailable`

on Nextcloud Desktop client.

On browser, it can be downloaded by: **Select it > Actions > Download**

## Logs

```
{
  "reqId":"8aBIAWaqMnQem77CAfpz",
  "level":3,
  "time":"2020-07-28T12:23:07+02:00",
  "remoteAddr":"1.2.3.4",
  "user":"testing",
  "app":"PHP",
  "method":"COPY",
  "url":"/remote.php/dav/files/testing/Dolphin%20Emulator/Logs/dolphin.log",
  "message":"fopen(httpseek://): failed to open stream: &quot;OC\\Files\\Stream\\SeekableHttpStream::stream_open&quot; call failed at /var/www/testing/lib/private/Files/Stream/SeekableHttpStream.php#67",
  "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0",
  "version":"19.0.1.1"
}{ 
  "reqId":"8aBIAWaqMnQem77CAfpz",
  "level":4,
  "time":"2020-07-28T12:23:07+02:00",
  "remoteAddr":"1.2.3.4",
  "user":"testing",
  "app":"webdav",
  "method":"COPY",
  "url":"/remote.php/dav/files/testing/Dolphin%20Emulator/Logs/dolphin.log",
  "message":{
    "Exception":"Sabre\\DAV\\Exception\\ServiceUnavailable",
    "Message":"Could not open file",
    "Code":0,
    "Trace":[      {
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Tree.php",
        "line":302,
        "function":"get",
        "class":"OCA\\DAV\\Connector\\Sabre\\File",
        "type":"->",
        "args":[

        
]
      
},
      {       
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Tree.php",
        "line":135,
        "function":"copyNode",
        "class":"Sabre\\DAV\\Tree",
        "type":"->",
        "args":[          {
            "__class__":"OCA\\DAV\\Connector\\Sabre\\File"
          
},
          {
            "__class__":"OCA\\DAV\\Files\\FilesHome"
          
},
          "dolphin.log"
        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line":687,
        "function":"copy",
        "class":"Sabre\\DAV\\Tree",
        "type":"->",
        "args":[
          "files/testing/Dolphin Emulator/Logs/dolphin.log",
          "files/testing/dolphin.log"
        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line":89,
        "function":"httpCopy",
        "class":"Sabre\\DAV\\CorePlugin",
        "type":"->",
        "args":[          {
            "__class__":"Sabre\\HTTP\\Request"
          
},
          {
            "__class__":"Sabre\\HTTP\\Response"
          
}
        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":474,
        "function":"emit",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[          "method:COPY",
          [            {
              "__class__":"Sabre\\HTTP\\Request"
            
},
            {
              "__class__":"Sabre\\HTTP\\Response"
            
}
          
]
        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":251,
        "function":"invokeMethod",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[          {
            "__class__":"Sabre\\HTTP\\Request"
          
},
          {
            "__class__":"Sabre\\HTTP\\Response"
          
}
        
]
      
},
      {       
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":319,
        "function":"start",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {        
        "file":"/var/www/testing/apps/dav/lib/Server.php",
        "line":320,
        "function":"exec",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {       
        "file":"/var/www/testing/apps/dav/appinfo/v2/remote.php",
        "line":35,
        "function":"exec",
        "class":"OCA\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {        
        "file":"/var/www/testing/remote.php",
        "line":167,
        "args":[
          "/var/www/testing/apps/dav/appinfo/v2/remote.php"
        
],
        "function":"require_once"
      
}
    
],
    "File":"/var/www/testing/apps/dav/lib/Connector/Sabre/File.php",
    "Line":428,
    "CustomMessage":"--"
  
},
  "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0",
  "version":"19.0.1.1"
}{
  "reqId":"W8FdAz7LcfdgpPbAFuzd",
  "level":3,
  "time":"2020-07-28T12:23:07+02:00",
  "remoteAddr":"1.2.3.4",
  "user":"testing",
  "app":"PHP",
  "method":"COPY",
  "url":"/remote.php/dav/files/testing/Dolphin%20Emulator/Logs/dolphin.log",
  "message":"fopen(http://10.1.0.2:8082/bucket-testing/urn%3Aoid%3A18377): failed to open stream: HTTP request failed! HTTP/1.1 416 Requested Range Not Satisfiable\r\n at /var/www/testing/lib/private/Files/ObjectStore/S3ObjectTrait.php#73",
  "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0",
  "version":"19.0.1.1"
}{
  "reqId":"W8FdAz7LcfdgpPbAFuzd",
  "level":3,
  "time":"2020-07-28T12:23:07+02:00",
  "remoteAddr":"1.2.3.4",
  "user":"testing",
  "app":"PHP",
  "method":"COPY",
  "url":"/remote.php/dav/files/testing/Dolphin%20Emulator/Logs/dolphin.log",
  "message":"fopen(httpseek://): failed to open stream: &quot;OC\\Files\\Stream\\SeekableHttpStream::stream_open&quot; call failed at /var/www/testing/lib/private/Files/Stream/SeekableHttpStream.php#67",
  "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0",
  "version":"19.0.1.1"
}{  
  "reqId":"W8FdAz7LcfdgpPbAFuzd",
  "level":4,
  "time":"2020-07-28T12:23:07+02:00",
  "remoteAddr":"1.2.3.4",
  "user":"testing",
  "app":"webdav",
  "method":"COPY",
  "url":"/remote.php/dav/files/testing/Dolphin%20Emulator/Logs/dolphin.log",
  "message":{    
    "Exception":"Sabre\\DAV\\Exception\\ServiceUnavailable",
    "Message":"Could not open file",
    "Code":0,
    "Trace":[      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Tree.php",
        "line":302,
        "function":"get",
        "class":"OCA\\DAV\\Connector\\Sabre\\File",
        "type":"->",
        "args":[

        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Tree.php",
        "line":135,
        "function":"copyNode",
        "class":"Sabre\\DAV\\Tree",
        "type":"->",
        "args":[          {
            "__class__":"OCA\\DAV\\Connector\\Sabre\\File"
          
},
          {
            "__class__":"OCA\\DAV\\Files\\FilesHome"
          
},
          "dolphin.log"
        
]
      
},
      {        
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line":687,
        "function":"copy",
        "class":"Sabre\\DAV\\Tree",
        "type":"->",
        "args":[
          "files/testing/Dolphin Emulator/Logs/dolphin.log",
          "files/testing/dolphin.log"
        
]
      
},
      {      
        "file":"/var/www/testing/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line":89,
        "function":"httpCopy",
        "class":"Sabre\\DAV\\CorePlugin",
        "type":"->",
        "args":[          {
            "__class__":"Sabre\\HTTP\\Request"
          
},
          {
            "__class__":"Sabre\\HTTP\\Response"
          
}
        
]
      
},
      {      
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":474,
        "function":"emit",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[          "method:COPY",
          [            {
              "__class__":"Sabre\\HTTP\\Request"
            
},
            {
              "__class__":"Sabre\\HTTP\\Response"
            
}
          
]
        
]
      
},
      {      
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":251,
        "function":"invokeMethod",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[          {
            "__class__":"Sabre\\HTTP\\Request"
          
},
          {
            "__class__":"Sabre\\HTTP\\Response"
          
}
        
]
      
},
      {      
        "file":"/var/www/testing/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line":319,
        "function":"start",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {      
        "file":"/var/www/testing/apps/dav/lib/Server.php",
        "line":320,
        "function":"exec",
        "class":"Sabre\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {      
        "file":"/var/www/testing/apps/dav/appinfo/v2/remote.php",
        "line":35,
        "function":"exec",
        "class":"OCA\\DAV\\Server",
        "type":"->",
        "args":[

        
]
      
},
      {     
        "file":"/var/www/testing/remote.php",
        "line":167,
        "args":[
          "/var/www/testing/apps/dav/appinfo/v2/remote.php"
        
],
        "function":"require_once"
      
}
    
],
    "File":"/var/www/testing/apps/dav/lib/Connector/Sabre/File.php",
    "Line":428,
    "CustomMessage":"--"
  
},
  "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0",
  "version":"19.0.1.1"
}
```

**Comment by maaattt** [comment]
Same here on NC 19.0.1 with OS on primary storage. This happens on when trying to create a new note with the Notes app.

**Comment by gcommit** [comment]
Same here with NC 19.0.0 ... We have more than one instance and this appears on every instance. On some you can't even download the files after the sync produced errors the first time. 

And another "symptom" is, that on some instances the sync completely stops, when an error because of the 0kB data was produced...

Is there a Sync-Client version which is working fine?

**Comment by jdhirst1** [comment]
As noted in #22164, this issue only affects S3-compatible object storage. This has been confirmed by testing with OpenStack Swift without issues.

**Comment by Corinari** [comment]
as stated in #22437, we are also affected by this Bug. Being a bigger provider of Nextcloud, this is problem for us. 

If i should add any Logs or anything, please let me know.

**Comment by Corinari** [comment]
In one case, i had a massive lost in data with filenames including whitespace or (german) sound alterations (Umlaute). 
This was also speculated in older Community posts, but i cant find those right now. 

Anyone else experiencing similar cases?

**Comment by lauhayden** [comment]
@Corinari I have not seen files being *corrupted* to zero bytes - only ones that were originally zero bytes causing problems when syncing. I would expect that corruption would be a different bug, and worthy of a different issue.

**Comment by acsfer** [comment]
Agree, please open separate issue.

**Comment by acsfer** [comment]
Should be fixed with https://github.com/nextcloud/server/pull/22651 if not, please advise.

**Comment by smalchow** [comment]
@lauhayden and @Corinari i do not agree, we also have this problem, specifically, right now with a nextcloud 19.02 (centos with s3 storage) and nextcloud client 3.0.x on several OS. primarily the error started popping up after we started installing clients version 3.x.
But we are almost 100% sure that files were corrupted while or after the upload (while seems more likely) we experienced this problem with some many files where the users insist that the file was OK locally that we can be sure that it happened during upload.

So i would say that files were corrupted to 0 bytes while upload, unfortunately it is hard to check in the storage if the files are actually ok, since it is in chunks on S3.

This is a very serious problem, and i am going to have a few very unpleasant conversations in the company and i need to explain and ensure that this problem does not happen anymore.

How can we help with this, right now i have no idea where to start really
